node/pkg/p2p/watermark_test.go

```diff
@@ -0,0 +1,192 @@
+package p2p
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"fmt"
+	"github.com/certusone/wormhole/node/pkg/accountant"
+	node_common "github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/governor"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	"github.com/ethereum/go-ethereum/crypto"
+	p2pcrypto "github.com/libp2p/go-libp2p-core/crypto"
+	p2ppeer "github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p/p2p/net/connmgr"
+	"go.uber.org/zap"
+	"testing"
+	"time"
+)
+
+type G struct {
+	// arguments passed to p2p.New
+	obsvC                  chan *gossipv1.SignedObservation
+	obsvReqC               chan *gossipv1.ObservationRequest
+	obsvReqSendC           chan *gossipv1.ObservationRequest
+	sendC                  chan []byte
+	signedInC              chan *gossipv1.SignedVAAWithQuorum
+	priv                   p2pcrypto.PrivKey
+	gk                     *ecdsa.PrivateKey
+	gst                    *node_common.GuardianSetState
+	networkID              string
+	bootstrapPeers         string
+	nodeName               string
+	disableHeartbeatVerify bool
+	rootCtxCancel          context.CancelFunc
+	gov                    *governor.ChainGovernor
+	acct                   *accountant.Accountant
+	signedGovCfg           chan *gossipv1.SignedChainGovernorConfig
+	signedGovSt            chan *gossipv1.SignedChainGovernorStatus
+	components             *Components
+}
+
+func NewG(nodeName string) *G {
+	cs := 20
+	p2ppriv, _, err := p2pcrypto.GenerateKeyPair(p2pcrypto.Ed25519, -1)
+	if err != nil {
+		panic(err)
+	}
+
+	guardianpriv, err := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+	if err != nil {
+		panic(err)
+	}
+
+	g := &G{
+		obsvC:                  make(chan *gossipv1.SignedObservation, cs),
+		obsvReqC:               make(chan *gossipv1.ObservationRequest, cs),
+		obsvReqSendC:           make(chan *gossipv1.ObservationRequest, cs),
+		sendC:                  make(chan []byte, cs),
+		signedInC:              make(chan *gossipv1.SignedVAAWithQuorum, cs),
+		priv:                   p2ppriv,
+		gk:                     guardianpriv,
+		gst:                    node_common.NewGuardianSetState(nil),
+		nodeName:               nodeName,
+		disableHeartbeatVerify: false,
+		rootCtxCancel:          nil,
+		gov:                    nil,
+		signedGovCfg:           make(chan *gossipv1.SignedChainGovernorConfig, cs),
+		signedGovSt:            make(chan *gossipv1.SignedChainGovernorStatus, cs),
+		components:             DefaultComponents(),
+	}
+
+	// Consume all output channels
+	go func() {
+		name := g.nodeName
+		fmt.Printf("[%s] consuming\n", name)
```

Comment:
```suggestion
		t.Logf("[%s] consuming\n", name)
```

If you are using `fmt.Printf()` in a test, that is generally a good indicator code smell. Please always avoid this an instead pass in a `*testing.T` and use [t.Logf()](https://pkg.go.dev/testing#T.Logf).

Tests should when feasible be as logic and code free as possible (this is hard in wormhole I know!).